ambry-server/src/main/java/com.github.ambry.server/VCRServer.java

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server;
+
+import com.codahale.metrics.JmxReporter;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.cloud.CloudBackupManager;
+import com.github.ambry.cloud.CloudDestinationFactory;
+import com.github.ambry.cloud.StaticVCRCluster;
+import com.github.ambry.clustermap.ClusterAgentsFactory;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.ClusterParticipant;
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.VirtualReplicatorCluster;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.ConnectionPoolConfig;
+import com.github.ambry.config.NetworkConfig;
+import com.github.ambry.config.ReplicationConfig;
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.config.ServerConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.network.BlockingChannelConnectionPool;
+import com.github.ambry.network.ConnectionPool;
+import com.github.ambry.network.NetworkServer;
+import com.github.ambry.network.Port;
+import com.github.ambry.network.PortType;
+import com.github.ambry.network.SocketServer;
+import com.github.ambry.notification.NotificationSystem;
+import com.github.ambry.store.FindTokenFactory;
+import com.github.ambry.store.StoreKeyConverterFactory;
+import com.github.ambry.store.StoreKeyFactory;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.utils.Utils.*;
+
+
+/**
+ * Virtual Cloud Replicator server
+ */
+public class VCRServer {
```

Comment:
VcrServer or VCRServer, which do you prefer?

Reply:
I prefer not to camel-case names that are acronyms, but I'll follow whatever our standard is.

Reply:
The suggested way is `VcrServer` (acronym longer than two characters), according to the class naming section of https://github.com/linkedin/ambry/wiki/Coding%20Guidelines

Reply:
Changed all to Vcr